govuk_chat_evaluation/rag_answers/custom_deepeval/metrics/context_relevancy/schema.py

```diff
@@ -0,0 +1,53 @@
+from typing import List, Literal
+from pydantic import BaseModel, Field
+
+
+class Verdict(BaseModel):
+    verdict: Literal["yes", "idk", "no"]
+    reason: str | None = Field(default=None)
+
+
+class VerdictCollection(BaseModel):
+    verdicts: List[Verdict]
+
+    def score_verdicts(self) -> float:
+        if len(self.verdicts) == 0:
+            return 1.0
+
+        quality_count = sum(1 for verdict in self.verdicts if verdict.verdict != "no")
+        return float(quality_count / len(self.verdicts))
+
+    def unmet_needs(self) -> List[str]:
+        unmet_needs: List[str] = []
```

Comment:
Do we have to specify a type here so that it knows the list isn't `List[str | None]`?